File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOWalkRemindTimesParamModel.swift
//
//  IDOWalkRemindTimesParamModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/18.
//

import Foundation

@objcMembers
public class IDOWalkRemindTimesParamModel: NSObject, IDOBaseModel {
    private let version: Int
    /// Switch: 1:On 0:Off
    public var onOff: Int
    private let num: Int
    public var items: [IDOWalkRemindTimesItem]

    enum CodingKeys: String, CodingKey {
        case version
        case onOff = "on_off"
        case num
        case items
    }

    public init(onOff: Int, items: [IDOWalkRemindTimesItem]) {
        self.version = 0
        self.onOff = onOff
        self.num = items.count
        self.items = items
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

// MARK: - IDOWalkRemindTimesItem
@objcMembers
public class IDOWalkRemindTimesItem: NSObject, Codable {
    /// Walk reminder time: Hour
    public var hour: Int
    /// Walk reminder time: Minute
    public var min: Int

    enum CodingKeys: String, CodingKey {
        case hour
        case min
    }

    public init(hour: Int, min: Int) {
        self.hour = hour
        self.min = min
    }
}

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOWalkRemindTimesParamModel.kt
//
//  IDOWalkRemindTimesParamModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.FieldNamingPolicy
import com.google.gson.GsonBuilder
import java.io.Serializable
import com.google.gson.annotations.SerializedName

///


open class IDOWalkRemindTimesParamModel(onOff: Int, items: List<IDOWalkRemindTimesItem>) :
    IDOBaseModel {

    @SerializedName("version")
    private var version: Int = 0

    /// Switch: 1:On 0:Off
    @SerializedName("on_off")
    var onOff: Int = onOff

    @SerializedName("num")
    private var num: Int = items.size

    @SerializedName("items")
    var items: List<IDOWalkRemindTimesItem> = items

    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}

// MARK: - IDOWalkRemindTimesItem
open class IDOWalkRemindTimesItem(hour: Int, min: Int) : Serializable {
    /// Walk reminder time: Hour
    @SerializedName("hour")
    var hour: Int = hour

    /// Walk reminder time: Minute
    @SerializedName("min")
    var min: Int = min

}